backend/src/api/play.rs

```diff
@@ -40,24 +60,55 @@ impl Game {
             .await
             .map_err(Error::from)?;
 
-        child.status().await?;
+        let (tx, rx) = mpsc::channel();
+        thread::spawn(move || {
+            let mut mov = String::new();
+            match listener.accept() {
+                Ok((mut socket, _)) => {
+                    let _ = socket.read_to_string(&mut mov);
+                },
+                Err(e) => println!("Failed to accept socket connection: {}", e),
+            }
+
+            let _ = tx.send(mov);
+        });
+        // Give 5s to compile and run the program
+        let mov = rx.recv_timeout(Duration::from_millis(5000));
+
+        // Kill the runner if it hasn't exited yet
+        if mov.is_err() {
```

Comment:
You could use a `let-else` here, since the Err case quits early, or a match if both paths are ~ the same length.